src/test/java/com/google/googleidentity/filter/UserAuthenticationFilterTest.java

```diff
@@ -90,9 +61,8 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
 
         verify(response).sendRedirect("/login");
 
-        assertEquals(
-                ((UserSession)request.getSession().getAttribute("user_session")).getOlduri().get(),
-                "/resource/user");
+        assertThat(((FakeHttpSession)request.getSession()).getUserSession().getOlduri())
+                .isEqualTo(Optional.ofNullable("/resource/user"));
```

Comment:
nit: use Truth8 OptionalSubject here.

Add `import static com.google.common.truth.Truth8.assertThat;` and then use `hasValue()`

```suggestion
        assertThat(request.getSession().getUserSession().getOlduri())
                .hasValue("/resource/user");
```